DDOT_module/parent_classes/marginals_analyzer_interface.py

```diff
@@ -0,0 +1,64 @@
+"""
+Created on 28.04.2021
+
+@author: fbm
+@contact: filip.b.maciejewski@gmail.com
+"""
+import abc
+
+"""
+    This file holds an interface for Marginals Analyzers. BS. 
+"""
+
+
+class MarginalsAnalyzerInterface(abc.ABC):
+    """
+    This is interface for classes that will analyze marginal probability distributions.
+    It requires those child classes to have basic functionalities that should be included.
+    for analyzing marginals.
+    """
+
+    @property
+    @abc.abstractmethod
+    def results_dictionary(self) -> dict:
```

Comment:
Do you expect the `dicts` in different `marginal analyzers` to be of different forms? If not, the do use `typing Dict` to specify the type.

Reply:
Yes. The specified versions are in MarginalsAnalyzerBase.